hw/xbox/mcpx_apu.c

```diff
@@ -66,7 +68,23 @@
 #define NV_PAPU_CVLMP                                    0x00002070
 #define NV_PAPU_NVLMP                                    0x00002074
 #define NV_PAPU_GPSMAXSGE                                0x000020D4
+#define NV_PAPU_GPFMAXSGE                                0x000020D8
 #define NV_PAPU_EPSMAXSGE                                0x000020DC
+#define NV_PAPU_EPFMAXSGE                                0x000020E0
+
+#define NV_PAPU_GPOFBASE0                                0x00003024
+#define NV_PAPU_GPOFEND0                                 0x00003028
+#define NV_PAPU_GPOFCUR0                                 0x0000302C
+#define NV_PAPU_GPIFBASE0                                0x00003064
+#define NV_PAPU_GPIFEND0                                 0x00003068
+#define NV_PAPU_GPIFCUR0                                 0x0000306C
```

Comment:
I don't like the formatting of these blocks. There should be a comment saying that there's 4 output FIFOs and 2 input FIFOs, with a stride of 0x10 bytes between the register sets.

I tried to find a similar case in nv2a but was unable to find a solution. Maybe someone else can find one?

Suggestions how to word this?

Reply:
I've seen this construct:
```
#define NV_PAPU_GPOFCUR(i)                                 (0x0000302C+((i)*0x10))
#define NV_PAPU_GPIFBASE(i)                                (0x00003064+((i)*0x10))
#define NV_PAPU_GPIFEND(i)                                 (0x00003068+((i)*0x10))
#define NV_PAPU_GPIFCUR(i)                                 (0x0000306C+((i)*0x10))
```

Possibly, add and use these?
```
#define NV_PAPU_GP_STRIDE 0x10
#define NV_PAPU_GP_COUNT 4
```